### DISCUSSION TITLE
Homing and leveling with Z probe

### DISCUSSION THREAD

**Discussion by asward** [grblHAL/core#308] [Q&A] [answered]
My machine has a ganged Z axis. Is it possible to use the Z probe to measure and level the z gantry with the existing core commands? If not, is there a plugin?

**Comment by andrewmarles** [comment, marked as answer]
Squaring of ganged axes is usually via the autosquaring homing function and ganged axis offsets.  You could use existing probe functions to determine the required offsets assuming you have a perfectly square reference surface.  Then the axis would be squared whenever you home.

If you are trying to correct for skew in the controller I don't think this is currently supported.  Surface mapping in GRBLHAL is accomplished by modifying the g-code in the sender.  Notably, if you perform a surface map with a skewed axis and then apply that map to a loaded g-code program it will inherently correct for a skewed Z axis.

**Reply by asward** [reply]
Thanks for the quick reply!

Marlin had a feature where you can successively probe in one axis to level the ganged (dual in marlin speak) axis to the work table, each iteration removing differential between the steppers until there is none. It's kind of a PITA with a probe that isn't automatically deployed/retracted (this is for a router so the probe is the bit I have installed). However, it was the only way I knew to do it.

I have since found a way to get sensorless homing to work on my lead screw Z axis, and have set the stops to what I hope will be acceptable. I can then probe to set a zero reference before starting working.

Again, appreciate your reply - it set me on a new research path and I found a good solution!
